kite-data/kite-data-crunch/src/main/java/org/kitesdk/data/crunch/CrunchDatasets.java

```diff
@@ -231,6 +232,38 @@ public static Target asTarget(URI uri) {
     }
   }
 
+  public static <E, K> PCollection<E> partitionAndSort(PCollection<E> collection,
+      View<E> view, MapFn<E, GenericData.Record> getSortKeyFn, Schema sortKeySchema) {
+    return partitionAndSort(collection, view, getSortKeyFn, sortKeySchema, -1);
+  }
+
+  public static <E, K> PCollection<E> partitionAndSort(PCollection<E> collection,
+      View<E> view, MapFn<E, GenericData.Record> getSortKeyFn, Schema sortKeySchema,
+      int numWriters) {
+    return partitionAndSort(collection, view, getSortKeyFn, sortKeySchema, numWriters, 1);
+  }
+
+  public static <E, K> PCollection<E> partitionAndSort(PCollection<E> collection,
+      View<E> view, MapFn<E, GenericData.Record> getSortKeyFn, Schema sortKeySchema,
```

Comment:
Can this be something other than GenericData.Record? It would be nice to parameterize this and allow the caller to pass in a PType<S> for the sort key rather than a Schema. There are a lot of cases where the caller just wants to extract a single field for the sort.

Speaking of extracting a single field, that may be a valuable variant as well: `partitionAndSort(PCollection, View, String fieldName)`  That doesn't need to go into this commit though.